src/realm/sync/client_syncsocket.hpp

```diff
@@ -0,0 +1,229 @@
+///////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef REALM_SYNC_CLIENT_SYNC_SOCKET_HPP
+#define REALM_SYNC_CLIENT_SYNC_SOCKET_HPP
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include <realm/status.hpp>
+
+#include <realm/sync/config.hpp>
+
+#include <realm/util/functional.hpp>
+#include <realm/util/optional.hpp>
+#include <realm/util/span.hpp>
+
+namespace realm::sync {
+
+/// Function handler typedef
+using FunctionHandler = util::UniqueFunction<void(Status)>;
```

Comment:
can you put this inside the SyncSocketInterface?

Reply:
done